server/src/main/java/org/apache/iotdb/db/query/dataset/NonAlignEngineDataSet.java

```diff
@@ -286,6 +293,11 @@ public TSQueryNonAlignDataSet fillBuffer(int fetchSize, WatermarkEncoder encoder
 
     for (int seriesIndex = 0; seriesIndex < seriesNum; seriesIndex++) {
       if (!noMoreDataInQueueArray[seriesIndex]) {
+        // check the interrupted status of main thread before take next batch
+        if (Thread.interrupted()) {
```

Comment:
The static method `interrupted()` in `Thread` class will clear the interrupt flag, you should call the instance method `isInterrupted()`.